uc-crux-llvm/src/UCCrux/LLVM/Soundness.hs

```diff
@@ -51,6 +54,14 @@ stringToSoundness =
     "indefinite" -> Just Indefinite
     _ -> Nothing
 
+soundnessToString :: Soundness -> String
+soundnessToString =
+  \case
+    Precise -> "precise"
+    Overapprox -> "overapprox"
+    Underapprox -> "underapprox"
```

Comment:
Aesthetics opinion: if these strings are displayed to users, I'd recommend full rather than abbreviated text (e.g. "over-approximation" v.s. "overapprox".

Is there a test somewhere that `stringToSoundness . soundnessToString == id` or is this not a needed invariant?

Reply:
> Aesthetics opinion: if these strings are displayed to users, I'd recommend full rather than abbreviated text (e.g. "over-approximation" v.s. "overapprox".

Ah, these strings are *written by* users. I chose these abbreviations because they're fairly unambiguous. The corresponding printing doesn't use these abbreviations.

> Is there a test somewhere that stringToSoundness . soundnessToString == id or is this not a needed invariant?

Documented. Since they're right next to each other and only four constructors, I'll skip a test for now...